build_magic/output.py

```diff
@@ -190,6 +193,18 @@ def info(self, msg):
         message = f'{datetime.now().isoformat()} build-magic [ INFO  ] OUTPUT   : {msg}'
         self._display(message)
 
+    def process_spinner(self, spinner, process_active=False):
```

Comment:
I feel like the Basic output shouldn't have any interactive notifications since the idea behind it is to provide log-like output for CI/CD tools. Did you manual test out the spinner in a non-terminal environment such as SSH, python's subprocess, or a shell script?

My gut tells me to take it out of the Basic output and just return None like `Basic.macro_start()`.